waiter/test/waiter/auth/oidc_test.clj

```diff
@@ -696,3 +696,33 @@
+           {:cookie-string "a=b" :expected "a=b"}
+           {:cookie-string "a=b; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-A1B2C3=1234; a=b" :expected "a=b"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-A1B2C3=1234" :expected "a=b"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=1234; a=b" :expected "a=b"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=1234" :expected "a=b"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=1234" :expected "a=b"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321; a=b" :expected "a=b"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; a=b; x-waiter-oidc-challenge-3cb2a1=321" :expected "a=b"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321" :expected "a=b"}
+           {:cookie-string "x-waiter-oidc-challenge-A1B2C3=1234; a=b; c=d" :expected "a=b; c=d"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-A1B2C3=1234; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=1234; a=b; c=d" :expected "a=b; c=d"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=1234; c=d" :expected "a=b; c=d"}
+           {:cookie-string "a=b; c=d; x-waiter-oidc-challenge-a1b2c3=1234" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321; a=b; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; a=b; x-waiter-oidc-challenge-3cb2a1=321; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; a=b; c=d; x-waiter-oidc-challenge-3cb2a1=321" :expected "a=b; c=d"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321; c=d" :expected "a=b; c=d"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=123; c=d; x-waiter-oidc-challenge-3cb2a1=321" :expected "a=b; c=d"}
+           {:cookie-string "a=b; c=d; x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-4d5e6f=456; x-waiter-oidc-challenge-3cb2a1=321; a=b; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; a=b; x-waiter-oidc-challenge-3cb2a1=321; x-waiter-oidc-challenge-4d5e6f=456; c=d" :expected "a=b; c=d"}
+           {:cookie-string "x-waiter-oidc-challenge-a1b2c3=123; a=b; c=d; x-waiter-oidc-challenge-3cb2a1=321; x-waiter-oidc-challenge-4d5e6f=456" :expected "a=b; c=d"}
+           {:cookie-string "a=b; x-waiter-oidc-challenge-a1b2c3=123; x-waiter-oidc-challenge-3cb2a1=321; c=d; x-waiter-oidc-challenge-4d5e6f=456" :expected "a=b; c=d"}]]
+    (is (= expected (auth/remove-auth-cookie cookie-string)) (str test-case))))
```

Comment:
I don’t see existing cases for `remove-auth-cookie` that tea removal of the other prefixes. Maybe that’s tested indirectly? Should they be tested explicitly like these oidc cases?

I might also include cases where the prefix string is present but it’s not used as a prefix, eg `blah-x-waiter-oidc-challenge-abc=123` should not be removed.

Reply:
Maybe include a case with an empty cookie string, too.

Reply:
Added additional test cases and a separate unit test function.